CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Contributing
+
+## Development
+
+### System Requirements
+
+Python 3.8 and above are required.
+
+### Target version(s)
+
+Python 3.8 and above are supported by the SDK.
+
+### Installation and Dependencies
+
+A [`Makefile`](./Makefile) has been included in the project which should make it straightforward to start the project locally. We utilize virtual environments (see [`virtualenv`](https://docs.python.org/3/tutorial/venv.html)) in order to provide isolated development environments for the project. This reduces the risk of invalid or corrupt global packages. It also integrates nicely with Make, which will detect changes in the `requirements-dev.txt` file and update the virtual environment if any occur.
```

Comment:
Is using something like pipenv not common? I've been out of the python world for a few years now.. but this is what I've been using in my projects. Handles generating lock files, contains scripts, etc. 

https://pipenv.pypa.io/en/latest/index.html#

Reply:
Either works. I haven't used `pipenv` yet but it seems like it would work well. I don't really have the time to research and change this all to use `pipenv` instead, so I created [an issue](https://github.com/open-feature/python-sdk/issues/59) to do so. If we prefer not to use Make I'll drop the code instead.

Reply:
I have a chip on my shoulder regarding pipenv as it went ~2 years unmaintained and was buggy as heck. It looks like it's maintained again now but my grudge is also well maintained 😄 

If we want to move to a dedicated packaging product, I would prefer [poetry](https://github.com/python-poetry/poetry). I'll follow up on the issue you created @tcarrio .

Reply:
I brought this up in the main thread [here](https://github.com/open-feature/python-sdk/pull/31#issuecomment-1334767068):

> The last outstanding work for this review seems to be utilizing a different build tool besides Make. I created [a separate issue](https://github.com/open-feature/python-sdk/issues/59) to track that work.
> 
> In the meantime would we prefer to remove Make until Poetry/pipenv/etc. is implemented or leave it to be replaced when that issue is taken on?

Reply:
I'm fine with keeping the makefile for now and dealing with the pipenv / poetry / etc. refactor later.